test/TurnUITests.java

```diff
@@ -2,29 +2,99 @@
 import com.TTT.BoardPrinter;
 import com.TTT.Game;
 import com.TTT.TurnUI;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class TurnUITests {
+
+    private Board board;
+    private Game game;
+    private BoardPrinter printer;
+
+    @Before
+    public void setUp() throws Exception {
+       board = new Board();
+       game = new Game(board);
+       printer = new BoardPrinter(System.out);
+    }
+
     @Test
     public void canAlternateTurns() {
-        TurnUI turns = new TurnUI();
-        Board board = new Board();
-        BoardPrinter printer = new BoardPrinter(System.out);
-        Game game = new Game(board);
-        ByteArrayInputStream in = new ByteArrayInputStream("1".getBytes());
+        ByteArrayInputStream in = new ByteArrayInputStream("1\n2\n".getBytes());
+        TurnUI turns = new TurnUI(printer, System.out, in);
+
+        turns.takeTurn(board, game);
+        turns.takeTurn(board, game);
+
+        assertEquals(board.markerAt(1), "O");
+    }
+
+    @Test
+    public void asksForInputAgainIfInputIsText() {
```

Comment:
Could you make these tests smaller? There is a ton of setup for each of these cases, each of which test the validity of the input. Why do you need to test this through the UI? I would want exactly one test that verifies that the user is asked again if the input in invalid, not one test for each possible type of invalid input. You still need those tests, but they shouldn't go through the UI.